### DISCUSSION TITLE
docket_id issues in Bulk Data

### DISCUSSION THREAD

**Discussion by PassusFrollo** [freelawproject/courtlistener#3047] [Q&A] [answered]
I am trying to work with some bulk data files, however I have noticed that the `docket_id` in opinion-clusters-2023-07-31.csv does not seem to match the `id` column in dockets-2023-07-31.csv.  

Has anyone ever encountered the same issue?  I am thinking of potentially trying some workarounds involving the slugs and case names, but those columns are not completely unique.  I have also noticed that the API does not have the same issue, as if I put the `docket_id` from the clusters table into an id search on the dockets endpoint it comes up with the correct case.  

This makes me think the ids in the dockets bulk data file are wrong (or I did something wrong). I am checking older dockets bulk data files to see if I have the same issue with them.

**Comment by mlissner** [comment]
It'd be rather shocking if the data were wrong, but you haven't said why you think this is true. Why do you think the docket_id field from clusters doesn't match the id field in dockets?

**Reply by PassusFrollo** [reply]
Actually, I may have found the problem.  Are there more than 66 million dockets in the bulk data file? That is far more than I was anticipating.  I may just need to rewrite some code to handle that quantity.

**Reply by mlissner** [reply, marked as answer]
Not sure, but a `wc -l` of the docket CSV should give an answer.

**Reply by PassusFrollo** [reply]
Yep, that works.  Thanks!
